rubert/src/pipeline.rs

```diff
@@ -32,6 +33,27 @@ pub enum PipelineError {
     Pooler(#[from] PoolerError),
 }
 
+impl<K, P> Pipeline<K, P>
+where
+    K: BertModel,
+{
+    pub fn from(config: Configuration<K, P>) -> Result<Self, PipelineError> {
+        let tokenizer = Tokenizer::new(
+            config.vocab,
+            config.accents,
+            config.lowercase,
+            config.token_size,
+        )?;
+        let model = Model::new(config.model, config.token_size)?;
```

Comment:
We can also map those errors to those we had in the builder 
```rust
    /// Failed to build the tokenizer: {0}
    Tokenizer(#[from] TokenizerError),
    /// Failed to build the model: {0}
    Model(#[from] ModelError),
```
this way we also keep the old error messages in `xayn-ai-ffi-c/src/reranker/ai.rs`

Reply:
We could, but since the pipeline can also return errors from its `run` methods, they would be slightly off in one of the scenarios in any case.

Reply:
What I meant was adding them as additional variants

```rust
pub enum PipelineError {
    /// Failed to run the tokenizer: {0}
    Tokenizer(#[from] TokenizerError),
    /// Failed to run the model: {0}
    Model(#[from] ModelError),
    /// Failed to run the pooler: {0}
    Pooler(#[from] PoolerError),
    /// Failed to build the tokenizer: {0}
    TokenizerBuild(#[source] TokenizerError),
    /// Failed to build the model: {0}
    ModelBuild(#[source] ModelError),
}

impl<K, P> Pipeline<K, P>
where
    K: BertModel,
{
    pub fn from(config: Configuration<K, P>) -> Result<Self, PipelineError> {
        let tokenizer = Tokenizer::new(
            config.vocab,
            config.accents,
            config.lowercase,
            config.token_size,
        )
        .map_err(PipelineError::TokenizerBuild)?;
        let model =
            Model::new(config.model, config.token_size).map_err(PipelineError::ModelBuild)?;

        Ok(Pipeline {
            tokenizer,
            model,
            pooler: config.pooler,
        })
    }
}
```